sos/plugins/crash_report.py

```diff
@@ -0,0 +1,256 @@
+    """Crash kernel report
+    """
+    plugin_name = "crash_report"
+    profiles = ('system', 'debug')
+
+    dbg_vmlinux_path = '/usr/lib/debug/'
+    vmcore_stnd_path = '/var/crash'
+
+    dump_file_prefix = ''
+    crashed_kernel_name = ''
+    vmlinux_path = ''
+    message_to_log = ''
+    vmcore_file_path = ''
+    crash_script = ""
+    # crash result output file
+    crashed_output_file = '/var/log/os_crash_report.txt'
+
+    # list of crash command
+    crash_cmd_list = [
+                    ("sys", "\"SYSTEM INFORMATION:\""),
+                    ("help -m", "\"STACK LOCATION INFO:\""),
+                    ("bt", "\"STACK TRACE OF CURRENT CONTEXT:\""),
+                    ("bt -a", "\"STACK TRACES OF ACTIVE TASKS:\""),
+                    ("kmem -i", "\"MEMORY USAGE:\""),
+                    ("kmem -s", "\"KMALLOC SLAB DATA:\""),
+                    ("mod", "\"MODULES:\""),
+                    ("ps", "\"PROCESS STATUS:\""),
+                    ("log", "\"SYSTEM MESSAGE BUFFER:\""),
+                    ("files", "\"OPEN FILES OF CURRENT CONTEXT:\""),
+                    ("dev -p", "\"PCI DEVICE DATA:\""),
+                    ("runq", "\"RUN QUEUE TASK:\""),
+                    ("mach -o", "\"OPALMSG LOG:\""),
+                    ("irq -s", "\"DUMP KERNEL IRQ STATE:\"")]
+
+    """
+    parse crash_cmd_list and form crash_script which will
+    be given as an input to crash command
+    """
+    def form_crash_script(self):
+        for idx in range(len(self.crash_cmd_list)):
+            cmd = self.crash_cmd_list[idx][0]
+            header = self.crash_cmd_list[idx][1]
+            self.crash_script += "!echo %s >>{OUTPUT}\n" % header
+            self.crash_script += "!echo >> {OUTPUT}\n"
+            self.crash_script += "%s >> {OUTPUT}\n" % cmd
+            self.crash_script += "!echo >> {OUTPUT}\n"
+            self.crash_script += "!echo >> {OUTPUT}\n"
+        self.crash_script += "quit\n"
+        self.crash_script += "!echo >> {OUTPUT}\n"
+        self.crash_script += "!echo >> {OUTPUT}\n"
+
+    """
+    used to find dump for crashed kernel.
+    it compares the time of dump file with last crashed time.
+    return
+        True  : In case of success.
+        False : In case of failure.
+    """
+    def is_dump_for_crashing_kernel(self,  file_path, crashing_time):
+        file_creation_info = time.ctime(os.path.getctime(file_path))
+
+        """
+        get the actual time string using split on complete line
+        ctime returns string as 'Mon Mar 13 02:54:32 2017'
+        we are interested only in Month date hrs:min field
+        """
+        vmcore_creation_time = file_creation_info.rsplit(':', 1)[0].split(
+                                                          ' ', 1)[1]
+        t1 = datetime.strptime(vmcore_creation_time, "%b %d %H:%M")
+        t2 = datetime.strptime(crashing_time, "%b %d %H:%M")
+        difference = t1 - t2
+        return True if difference.days >= 0 else False
```

Comment:
Yes logic is correct.
I will add more and proper comment to explain things clearly.